### PR TITLE
Add support for automatically signing external modules

### PR DESCRIPTION
**Pull request by petronio**
Two new configuration options:
- **KERNEL_SIGNING_CERT**
- **KERNEL_SIGNING_KEY**

Possible values:
- unset
- "_/path/to/key/or/certificate_"
- "auto"

### CODE REVIEW

**Comment by sakaki-** [comment]
Thanks for this, I'll test it tonight, and merge tomorrow if it all works.
In the meantime please add version bump to your PR, as per `genup`.
Best, sakaki

**Comment by petronio** [comment]
No problem, version bumped. Thanks for reviewing.

File: buildkernel [hunk 1]
```diff
@@ -423,6 +424,29 @@ source_etc_conf_file() {
...
+            die "Cannot proceed; KERNEL_SIGNING_CERT is configured, but KERNEL_SIGNING_KEY is not."
+        fi
+        if [[ "${KERNEL_SIGNING_CERT}" == "auto" || "${KERNEL_SIGNING_KEY}" == "auto" ]]; then
+            if [[ "${KERNEL_SIGNING_CERT}" != "${KERNEL_SIGNING_KEY}" ]]; then
+                die "Cannot proceed; in automatic external module signing mode, both KERNEL_SIGNING_CERT and KERNEL_SIGNING_KEY must be set to \"auto\""
+            fi
+            KERNEL_SIGNING_CERT="${LINUXDIR}/certs/signing_key.x509"
+            KERNEL_SIGNING_KEY="${LINUXDIR}/certs/signing_key.pem"
+        else
+            if [[ ! -f KERNEL_SIGNING_CERT ]]; then
```
Comment (sakaki-) [inline buildkernel]: These `-f` tests should use `${VAR}` not `VAR`

Comment (petronio) [reply buildkernel]: Thanks, fixed.

**Comment by sakaki-** [comment]
Updated ebuild for this has now also been pushed.
Thanks again for the contribution ^-^
Best, sakaki

**Pull request merged**